cns/restserver/internalapi.go

```diff
@@ -151,6 +152,37 @@ func (service *HTTPRestService) SyncNodeStatus(dncEP, infraVnet, nodeID string,
 	return
 }
 
+// SyncHostNCVersion will check NC version from NMAgent and save it as host version in container status.
+// If NMAgent updated, CNS will refresh the pending programming IP status.
+func (service *HTTPRestService) SyncHostNCVersion(channelMode string) {
```

Comment:
Can you move this function out from internalapi.go? InternalApi is used for APIs exposed for CNS service for internal consumption. This is an API invoked for the background go routine like IpamPoolMonitor. Can you move it to a separate file like SyncNCProgrammingVersion.go?

Reply:
I can move it to a separate file but not sure whether we should do it. We can sync offline for this, I'm fine in both ways.
All HTTPRestService are listed in that file, include background thread SyncNodeStatus.
In other side, SyncHostNCVersion logic is not complex when compared to IpamPoolMonitor. Ipam pool monitor has 6 functions but SyncHostNCVersion only need get NC version and update it.